ntoskrnl/mm/section.c

```diff
@@ -3634,13 +3634,55 @@ MiRosUnmapViewOfSection(IN PEPROCESS Process,
     }
     else
     {
+        PMM_SECTION_SEGMENT Segment = MemoryArea->SectionData.Segment;
+        PMMVAD Vad = &MemoryArea->VadNode;
+        SIZE_T ViewSize = PAGE_SIZE + ((Vad->EndingVpn - Vad->StartingVpn) << PAGE_SHIFT);
+        LARGE_INTEGER ViewOffset;
+        ViewOffset.QuadPart = MemoryArea->SectionData.ViewOffset;
```

Comment:
Suggestion: Move the initializations of ViewSize and ViewOffset when they are needed (i.e. down at the `while (ViewSize > 0)` loop).

Reply:
Moved.